server/views/pages/analyticsBehaviourEntries.njk

```diff
@@ -19,7 +19,84 @@
 
       {{ dataSource("NOMIS", lastUpdated, {"classes": "govuk-!-margin-bottom-6"}) }}
 
-      CHART
+      <div>
+        <a href="?" class="govuk-button govuk-button--secondary govuk-!-margin-right-6">Download chart</a>
+
+        <div aria-hidden="true" class="app-chart-legend">
+          <span class="app-chart-legend__swatch app-chart-colour--a">&nbsp;</span>
+          <span class="app-chart-legend__label">Positive</span>
+          <span class="app-chart-legend__swatch app-chart-colour--b">&nbsp;</span>
+          <span class="app-chart-legend__label">Negative</span>
+        </div>
+      </div>
+
+      <table class="app-chart-table app-chart-table--bars">
+        <caption class="govuk-visually-hidden">
+          Comparison of positive and negative behaviour entries by wing in the last 28 days
+        </caption>
+
+        <thead>
+          <tr>
+            <th scope="col">Location</th>
+            <th scope="col">Positive <span class="govuk-visually-hidden">entries</span></th>
+            <th scope="col">Negative <span class="govuk-visually-hidden">entries</span></th>
+            <th aria-hidden="true" class="govuk-visually-hidden" scope="col">Chart</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {% for row in behaviourEntries %}
+            {% set totalEntries = row.entriesPositive + row.entriesNegative %}
+            {% set percentagePositive = row.entriesPositive | percentageOf(totalEntries) %}
+            {% set percentageNegative = row.entriesNegative | percentageOf(totalEntries) %}
+
+            <tr>
+              <td>
+                {% if row.href %}
+                  <a href="{{ row.href }}">{{ row.location }}</a>
+                {% else %}
+                  {{ row.location }}
+                {% endif %}
+              </td>
+              <td>
+                {{ percentagePositive }}
+                <br />
+                {{ row.entriesPositive | thousands }} <span class="govuk-visually-hidden">entries</span>
+              </td>
+              <td>
+                {{ percentageNegative }}
+                <br />
+                {{ row.entriesNegative | thousands}} <span class="govuk-visually-hidden">entries</span>
+              </td>
+              <td aria-hidden="true" class="app-chart-table__percentage-cell">
+              <span class="app-chart-table__axis-midpoint"></span>
+                <div id="c-pos-{{ loop.index }}" class="app-chart-table__percentage-cell-bar app-chart-colour--a">&nbsp;<br />&nbsp;</div>
+                <div id="c-neg-{{ loop.index }}" class="app-chart-table__percentage-cell-bar app-chart-colour--b">&nbsp;<br />&nbsp;</div>
+                <style nonce="{{ cspNonce }}">
+                  #c-pos-{{ loop.index }} { width:{{ percentagePositive }}; }
+                  #c-neg-{{ loop.index }} { width:{{ percentageNegative }}; }
```

Comment:
🧙

Reply:
CSP does not allow us to use style attributes. ugh. makes charts rather hard to do!
there are some options…